tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala

```diff
@@ -72,6 +74,73 @@ private[scalajs] final class ScalaJSClassEmitter(
 
   private[emitter] def semantics: Semantics = linkingUnit.semantics
 
+  private[emitter] lazy val typeTags: Map[LinkedClass, Int] = {
+    import Definitions.ReservedTags
+
+    val childrenMap =
+      linkingUnit.classDefs.groupBy(_.superClass.map(_.name)).withDefaultValue(Nil)
+
+    // Some classes may have parents that have been removed
+    val orphans = for(cls <- linkingUnit.classDefs;
+                    parent <- cls.superClass
+                    if !linkedClassByName.isDefinedAt(parent.name)) yield cls
```

Comment:
Prefer `contains()` over `isDefinedAt` for `Map`s. `isDefinedAt` implies working with a `PartialFunction`.